performance-tests/bench_2/worker/main.cpp

```diff
@@ -217,15 +217,14 @@ int ACE_TMAIN(int argc, ACE_TCHAR* argv[]) {
     std::mutex cv_mutex;
 
     if (!(config.create_time == ZERO)) {
+      std::chrono::time_point<std::chrono::high_resolution_clock> timeout_time;
       if (config.create_time < ZERO) {
-        auto dur = -get_duration(config.create_time);
-        std::unique_lock<std::mutex> lock(cv_mutex);
-        while (cv.wait_for(lock, dur) != std::cv_status::timeout) {}
+        timeout_time = std::chrono::high_resolution_clock::now() - get_duration(config.create_time);
       } else {
-        auto timeout_time = std::chrono::system_clock::time_point(get_duration(config.create_time));
-        std::unique_lock<std::mutex> lock(cv_mutex);
-        while (cv.wait_until(lock, timeout_time) != std::cv_status::timeout) {}
+        timeout_time = std::chrono::high_resolution_clock::time_point(get_duration(config.create_time));
       }
+      std::unique_lock<std::mutex> lock(cv_mutex);
+      while (cv.wait_until(lock, timeout_time) != std::cv_status::timeout) {}
```

Comment:
What is it actually waiting for?  The condition variable wait can't complete until some external state change, protected by the cv's mutex, has been observed.

Reply:
It's waiting for me to remember / look up how to just do a normal thread sleep in C++11. :)